### ISSUE TITLE
In data-view, after edit and saving, labels for all fields are visible, even if they have no values

### ISSUE THREAD

**Issue by Angamanga** [ushahidi/platform#2219]
### Expected behaviour
In data-view, when viewing a post in detail-mode after editing a post, only the fields that have values should be visible

### Actual behaviour
Labels for all fields are visible, even if they have no values

### Steps to reproduce the behaviour/error
1. go to data-view and select a post + select edit
2. delete the values for one of the fields
3. save
4. the label for the empty value is still visible:

![screen shot 2017-10-24 at 12 55 27](https://user-images.githubusercontent.com/8624777/31939510-eae2a698-b8ba-11e7-8e48-c265bfe3b678.png)

##### Where
- [x] [Local setup with Vagrant ](https://www.ushahidi.com/support/install-ushahidi#installing-for-development)
- [ ] [Local setup from platform-release ](https://www.ushahidi.com/support/install-ushahidi#installing-the-latest-release)
- [ ] Ushahidi.io / SaaS solution 
- [ ] Ushahidi's QA environment
- [x] Other (explain): sandbox.uchaguzi.or.ke

**Comment by crcommons** [comment]
Other fixes appear to fix this bug

**Comment by jshorland** [comment]
I am still seeing this happen in sandbox.uchaguzi.or.ke. 

See screenshot. 

<img width="1273" alt="screen shot 2017-10-24 at 1 39 16 pm" src="https://user-images.githubusercontent.com/3224856/31966836-d40dd0e0-b8c0-11e7-8a6b-d918c40e6d74.png">

It happened without me having to delete the values. Steps to replicate:
- I clicked on https://sandbox.uchaguzi.or.ke/posts/5824 and took it into edit mode
- I added 1 field "refine location name" and typed "somewhere" in that field
- I saved the post
- it took me to post detail view and I see all the fields even though they dont' have values. 
- I click on another post and then click on https://sandbox.uchaguzi.or.ke/posts/5824 AGAIN
- the post looks normal again, but it doesn't have the "refine location name" field data that I saved earlier.
